src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -68,8 +68,17 @@
 public class BitbucketSCMSource extends SCMSource {
 
     private static final Logger LOGGER = Logger.getLogger(BitbucketSCMSource.class.getName());
-    private CustomGitSCMSource gitSCMSource;
+    @UpgradeHandled(handledBy = "static fields not persisted", removeAnnotationInVersion = "4.0.0")
+    private static final String REFSPEC_DEFAULT = "+refs/heads/*:refs/remotes/@{remote}/*";
+    @UpgradeHandled(handledBy = "initialize() method", removeAnnotationInVersion = "4.0.0")
+    private String cloneUrl;
+    @SuppressWarnings("unused") // Kept for backward compatibility
+    private transient CustomGitSCMSource gitSCMSource;
+    @UpgradeHandled(handledBy = "initialize() method", removeAnnotationInVersion = "4.0.0")
```

Comment:
nit:
```suggestion
    @UpgradeHandled(handledBy = "readResolve() method", removeAnnotationInVersion = "4.0.0")
```

Reply:
But it is not.. read-resolve will throw an NPE when it calls `validateInitialized` 
Thus after re-start you will lose all your BitbucketSCMSource settings for the job.

Reply:
That's true. I've also [mentioned this in a separate comment](https://github.com/jenkinsci/atlassian-bitbucket-server-integration-plugin/pull/418#discussion_r1283890631) which I believe @atl-dyon has already addressed in a follow-up commit.